File: JavaCheatSheet/src/main/java/com/github/jameswdelancey/javaCheatSheet/Part3.java
package com.github.jameswdelancey.javaCheatSheet;

// Define an enumeration called Fruit
enum Fruit {
    Apple, Banana, Strawberry
}

// Define a type alias for Fruit
class Food {
    Fruit fruit;

    Food(Fruit fruit) {
        this.fruit = fruit;
    }
}

public class Part3 {
    public static void sayHello() {
        System.out.println("Hello World.");
    }

    public static void main(String[] args) {
        int x = 42;
        int y = -69;
        int xy = x * y;
        Tuple z = new Tuple(x, (float) y); // Tuple to store values of different types
        int[] arr = new int[] { x, y };
        int underscore = arr[0];

        // Define a Secrets class with fields x (double) and y (int)
        class Secrets {
            double x;
            int y;
        }

        Secrets secrets = new Secrets();
        secrets.x = -32.0;
        secrets.y = -42;

        Fruit fruit = Fruit.Apple;

        // Assign the sayHello method to a functional interface
        Runnable fun = Part3::sayHello;
        fun.run();

        System.out.println(x + " * " + y + " = " + xy);

        // Define a type alias for Fruit
        Food food = new Food(fruit);

        // Assign an empty tuple to a variable
        Object emptyTuple = new Object();
    }

    // Define a Tuple class to store values of different types
    static class Tuple {
        int x;
        float y;

        Tuple(int x, float y) {
            this.x = x;
            this.y = y;
        }
    }

    // Define a functional interface for the sayHello method
    @FunctionalInterface
    interface Runnable {
        void run();
    }

}

File: rustCheatSheet/src/part3.rs
fn sayHello() {
    println!("Hello World.")
}
#[test]
fn main() {
    let x = 42;
    let y = -69;
    let xy = x * y;
    let z = (x, y as f32); // tuple any types inside
    let _ = [x, y];
    let _ = z.0;
    let z = [x as f32, y as f32];
    let _ = z[0];

    struct Secrets {
        x: f64,
        y: i32,
    }
    let _ = Secrets { x: -32.0, y: -42 };

    enum Fruit {
        Apple,
        Banana,
        Strawberry,
    }
    let _ = Fruit::Apple;

    let fun = sayHello;
    fun();
    println!("{} * {} = {}", x, y, xy);
    type Food = Fruit;
    let _ = Food::Apple;
    let _ = ();
}
